test/samples.jl

```diff
@@ -0,0 +1,126 @@
+                encode!(tmp, d, nothing)
+                @test isapprox(tmp, s.data, rtol = 10)
+                @test isapprox(encode(d, missing).data, s.data, rtol = 10)
+                encode!(tmp, d, missing)
+                @test isapprox(tmp, s.data, rtol = 10)
+            else
+                tmp = similar(s.data)
+                encode!(tmp, d, nothing)
+                @test tmp == s.data
+                @test encode(d, nothing).data == s.data
+                encode!(tmp, d, missing)
+                @test isapprox(tmp, s.data, rtol = 1)
+                @test isapprox(encode(d, missing).data, s.data, rtol = 1)
+            end
+            chs = s.info.channels
+            i = 27
+            @test s[:, i].data == s.data[:, i:i]
+            t = TimeSpans.time_from_index(s.info.sample_rate, i)
+            t2 = TimeSpans.time_from_index(s.info.sample_rate, i + 15)
+            j = TimeSpans.index_from_time(s.info.sample_rate, t2) - 1
+            for ch_inds in (:, 1:2, 2:3, 1:3, [3,1], [2,3,1], [1,2,3])
+                @test s[chs[ch_inds], t].data == s[ch_inds, i].data
+                @test s[chs[ch_inds], TimeSpan(t, t2)].data == s.data[ch_inds, i:j]
+                @test s[chs[ch_inds], i:j].data == s.data[ch_inds, i:j]
+                @test s[ch_inds, t].data == s[ch_inds, i].data
+                @test s[ch_inds, TimeSpan(t, t2)].data == s.data[ch_inds, i:j]
+                @test s[ch_inds, i:j].data == s.data[ch_inds, i:j]
+            end
+            @test size(s[:, TimeSpan(0, Second(1))].data, 2) == floor(s.info.sample_rate)
+            for i in 1:length(chs)
+                @test channel(s, chs[i]) == i
+                @test channel(s, i) == chs[i]
+                @test channel(s.info, chs[i]) == i
+                @test channel(s.info, i) == chs[i]
+            end
+            @test s[:, TimeSpan(0, TimeSpans.duration(s))].data == s.data
+        end
+    end
+end
+
+@testset "`Samples` pretty printing" begin
+    info = SamplesInfo("eeg", ["a", "b", "c-d"], "unit", 0.25, -0.5, Int16, 50.2)
+    samples = Samples(rand(Random.MersenneTwister(0), info.sample_type, 3, 5), info, true)
+    @test sprint(show, samples, context=(:compact => true)) == "Samples(3×5 Array{Int16,2})"
+    @test sprint(show, samples) == """
+                                   Samples (00:00:00.099601594):
+                                     info.kind: "eeg"
+                                     info.channels: ["a", "b", "c-d"]
+                                     info.sample_unit: "unit"
+                                     info.sample_resolution_in_unit: 0.25
+                                     info.sample_offset_in_unit: -0.5
+                                     info.sample_type: Int16
+                                     info.sample_rate: 50.2 Hz
+                                     encoded: true
+                                     data:
+                                   3×5 Array{Int16,2}:
```

Comment:
FYI in 1.7 this will show as `Matrix{Int16}`, which will make this test fail.

Reply:
> FYI in 1.7 this will show as Matrix{Int16}, which will make this test fail.

Yup, that's why it's failing on nightly, which is allowed to fail